x/ecocredit/server/utils/utils.go

```diff
@@ -51,3 +50,27 @@ func GetNonNegativeFixedDecs(precision uint32, decimals ...string) ([]math.Dec,
 	}
 	return decs, nil
 }
+
+// GetCreditTypes fetches the credit type slice from param state.
+func GetCreditTypes(sdkCtx sdk.Context, pk ecocredit.ParamKeeper) []*core.CreditType {
+	var creditTypes []*core.CreditType
+	pk.Get(sdkCtx, core.KeyCreditTypes, &creditTypes)
```

Comment:
We should use `CreditTypeTable` for accessing credit type info. These params can be changed by governance proposal.

Reply:
we are still using the `x/params` module, https://github.com/regen-network/regen-ledger/issues/729 was moved for v5, so i think we should actually be removing the CreditType ORM table definition.

CreditTypes are automatically saved via gov param change proposals and the x/params module. Come to think of it though, it might actually be a good idea to save the credit type into the ORM table during class creation for quicker fetching.. Maybe we can discuss during next standup?

Reply:
> CreditTypes are automatically saved via gov param change proposals and the x/params module. Come to think of it though, it might actually be a good idea to save the credit type into the ORM table during class creation for quicker fetching.. Maybe we can discuss during next standup?


Yes, We should save credit type info during class creation. If credit type is removed by gov param change, we can't issue create credit batches.